src/wallet/wallet_tx_builder.cpp

```diff
@@ -446,6 +525,92 @@ WalletTxBuilder::IterateLimit(
     return std::make_tuple(spendableMut, updatedFee, changeAddr);
 }
 
+static tl::expected<ResolvedPayment, AddressResolutionError>
+ResolvePayment(
+        const Payment& payment,
+        bool canResolveOrchard,
+        const TransactionStrategy& strategy,
+        CAmount& maxSaplingAvailable,
+        CAmount& maxOrchardAvailable,
+        uint32_t& orchardOutputs)
+{
+    return examine(payment.GetAddress(), match {
+        [&](const CKeyID& p2pkh) -> tl::expected<ResolvedPayment, AddressResolutionError> {
+            if (strategy.AllowRevealedRecipients()) {
+                return {{std::nullopt, p2pkh, payment.GetAmount(), payment.GetMemo(), false}};
```

Comment:
nonblocking: The lack of an explicit constructor call squicks me a little, given that we have std::variant in the mix.